### ISSUE TITLE
Anyway to add new modules?

### ISSUE THREAD

**Issue by eminarcissus** [omz/PythonistaAppTemplate#11]
In a native pythonista implementation we cannot add cython modules because we don't have a c compiler in iOS runtime. Is there anyway to add a cython module to a Xcode project to pre compile the module and have it linked in the project?

**Comment by Cerise1** [comment]
Hello emin, 
I have the exact same problem... I would like to create a standalone app but my script need 3 more modules to work properly : twisted, PyNacl( require libsodium a C librairies) and pyblake2 ! 
Did you find a solution to link thoose modules in the project ?

**Comment by eminarcissus** [comment]
Have no way out until now.
I guess Pythonista itself is a self contained lib, so there's no way to extend our c lib unless re-compile python with it.

**Comment by Cerise1** [comment]
And do you think its possible to recompose Python with it ? 
I am stuck on my project be cause of that ! I thought it was a great way to run Python 3.5 code in an iOS app as i dont know other way to install Python 3.5 on the idevice even if its jailbroken.. And i cant use Python 2.. :(

**Comment by controversial** [comment]
Pythonista isn't a platform to magically make your Python scripts run as an iOS app. You'll have to compile the C libraries yourself and link them with Python. You'll probably need more technical knowledge than it sounds like you have

**Comment by Cerise1** [comment]
Yes i agree with you i need more Technical knowledge to do that but i ve been searching for 3 days on the Internet how to manualy compile and link C lib with Python i dont find anything ! I am sure its long to explain but could you redirect me some where i could learn that or could you Tell me what should i learn i am ok to work :) 
Or if you have a better idea to run Python 3.5 code in iOS app i would be gratefull :)

**Comment by disorientedperson** [comment]
The easiest thing to do is wait until @omz updates the template.

**Comment by controversial** [comment]
Right. This template does not yet support Python 3. 

You *can* install Twisted by copying its files to `site-packages`, I think you'll need `zope.interface`first as well. You're out of luck for libsodium though. Could you rewrite your code not to use it?

**Comment by Cerise1** [comment]
The default script is written in Python 3 ( print('hello world') ) with the () ! You sure that it doesnt support Python 3 ? 

Yes you are right i had the zope.interface error when i tested to add twisted file manualy ! 
No i cant i woudnt be able to re write all its very long and complex : a lot of thing are on top of pyNacl.. :(

**Comment by disorientedperson** [comment]
I'm pretty sure it doesn't. That's just good practice for when it _is_ ready for Python 3 since that's backwards-compatible.

**Comment by Cerise1** [comment]
Why the default script main.py would be written in Python 3 ( with print interprètes as a function with parenthesis) if the template is not compatible with Python 3 though ?

**Comment by controversial** [comment]
@Cerise1 That's perfectly valid Python 2 code.
